server/tournaments/index.js

```diff
@@ -9,42 +9,51 @@ const TOURBAN_DURATION = 14 * 24 * 60 * 60 * 1000;
 
 Punishments.roomPunishmentTypes.set('TOURBAN', 'banned from tournaments');
 
+/** @type {{[k: string]: Object}} */
 let TournamentGenerators = Object.create(null);
-let generatorFiles = {
+/** @type {{roundrobin: string, elimination: string}} */
+const generatorFiles = {
 	'roundrobin': 'generator-round-robin',
 	'elimination': 'generator-elimination',
 };
 for (const type in generatorFiles) {
+	// @ts-ignore
 	TournamentGenerators[type] = require('./' + generatorFiles[type]);
 }
 
+/** @type {{[k: string]: Tournament}} */
 exports.tournaments = {};
 
+/** @param {User[]} users */
 function usersToNames(users) {
 	return users.map(user => user.name);
 }
 
-class Tournament {
+class Tournament extends Rooms.RoomGame {
+	/**
+	 * @param {ChatRoom} room
+	 * @param {Format} format
+	 * @param {Object} generator TODO: type this properly
+	 * @param {string | undefined} playerCap
+	 * @param {boolean} isRated
+	 */
 	constructor(room, format, generator, playerCap, isRated) {
-		if (room.tourNumber) {
-			room.tourNumber++;
```

Comment:
You seem to have removed the `tourNumber` tracker without adding it back in elsewhere? Is it useless or something?

Reply:
according to gentle observation and ``rg``, yes, it used to be appended to `teambuilderFormat`